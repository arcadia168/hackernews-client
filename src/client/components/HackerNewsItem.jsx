import React, { Component } from 'React';
import { Row } from 'react-bootstrap';
import moment from 'moment';

// TODO: Use a pure functional component if no state changes...
export default class HackerNewsItem extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        debugger;
        console.info(`In hacker news item, the props are: ${JSON.stringify(this.props)}`);

        //by: "jamesbowman"
        // descendants: 14
        // id: 20397044
        // kids: Array(7) [ 20397310, 20397590, 20397347, â€¦ ]
        // score:66
        // time: 1562709157
        // title: "I have found an excellent programmer named Steve Wozniac"
        // type: "story"
        // url: "https://tinyletter.com/jamesbowman/letters/i-have-found-an-excellent-programmer-named-steve-wozniac"

        // Calculate 'hours ago' from now...
        // const momentTimePosted = moment.unix(this.props.itemDetails.time).format('HH');
        // console.info(`The momentTimePosted (in hours): ${momentTimePosted}`);
        // debugger;
        // const momentFromNow = momentTimePosted.fromNow();
        // debugger;

        return (
            <Row>
                <h1>{this.props.itemDetails.title}</h1>
                <h2>{this.props.itemDetails.score} by {this.props.itemDetails.by} </h2>
            </Row>
        );
    }
}
