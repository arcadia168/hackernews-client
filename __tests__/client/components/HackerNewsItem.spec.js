import React from 'react';
import HackerNewsItem from '../../../src/client/components/HackerNewsItem';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('App', () => {
    const render = customProps => {
        const props = {
            // Default props
            ...customProps,
        }
        return mount(<App {...props} />);
    }

    it('renders the app as expected', () => {
        const testItemDetails = {
            "by": "dhouston",
            "descendants": 71,
            "id": 8863,
            "kids": [8952, 9224, 8917, 8884, 8887, 8943, 8869, 8958, 9005, 9671, 8940, 9067, 8908, 9055, 8865, 8881, 8872, 8873, 8955, 10403, 8903, 8928, 9125, 8998, 8901, 8902, 8907, 8894, 8878, 8870, 8980, 8934, 8876],
            "score": 111,
            "time": 1175714200,
            "title": "My YC app: Dropbox - Throw away your USB drive",
            "type": "story",
            "url": "http://www.getdropbox.com/u/2/screencast.html"
        }
        const component = renderer.create(
            <HackerNewsItem itemDetails={testItemDetails} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

