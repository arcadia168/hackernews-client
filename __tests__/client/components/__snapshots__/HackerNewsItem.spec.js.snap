// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app as expected 1`] = `
<div
  className="hacker-news-item__container row"
>
  <h5
    className="hacker-news-item__title"
  >
    <a
      href="http://www.getdropbox.com/u/2/screencast.html"
    >
      My YC app: Dropbox - Throw away your USB drive
    </a>
  </h5>
  <div
    className="hacker-news-item__details"
  >
    <span
      className="hacker-news-item__description"
    >
      111
       by 
      dhouston
       at 
      04/04/2007 20:16
    </span>
    <button
      className="hacker-news-item__comment-btn btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Show comments
    </button>
  </div>
</div>
`;
